zookeeper-server/src/main/java/org/apache/zookeeper/server/ZooKeeperServer.java

```diff
@@ -200,6 +202,7 @@ void removeCnxn(ServerCnxn cnxn) {
     public ZooKeeperServer() {
         listener = new ZooKeeperServerListenerImpl(this);
         serverStats = new ServerStats(this);
+        this.requestPathMetricsCollector = new RequestPathMetricsCollector();
```

Comment:
what about having a default no-op implementation of RequestPathMetricsCollector and make this feature disabled by default ?
it would consume CPU cycles even if not needed

Reply:
introduced a flag `enabled` with default value `false`. Removed the old `disabled` flag.